datajoint/condition.py

```diff
@@ -183,6 +190,10 @@ def combine_conditions(negate, conditions):
             return not negate  # and empty AndList is True
         return combine_conditions(negate, conditions=items)
 
+    # restrict by Top
+    if isinstance(condition, Top):
+        return query_expression.make_subquery(top_restriction=condition)
```

Comment:
Should we pass `top_restriction` as a function argument or treat it as other restrictions, i.e. a member of the `QueryExpression` class?

Reply:
Perhaps even just add it as the last member of the `self.restrictions` list.

Reply:
I am already setting it as an attribute of QueryExpression within `make_subquery`, but I can simply set it first and then make the subquery, removing the need for a `top_restriction' argument. It makes more sense to me to have it as its own attribute rather than including it in the `restriction` list.

Reply:
I take my comment back. I think we should intercept `dj.Top` in `QueryExpression.restrict`.